src/app/public/modules/phone-field/phone-field-input.directive.ts

```diff
@@ -0,0 +1,240 @@
+      this.skyPhoneFieldComponent.selectedCountry.placeholder);
+
+    this.adapterService.setAriaLabel(this.elRef.nativeElement);
+  }
+
+  public ngOnDestroy(): void {
+    this.ngUnsubscribe.next();
+    this.ngUnsubscribe.complete();
+    this.selectedCountryChange.complete();
+  }
+
+  public ngAfterViewInit(): void {
+    this.skyPhoneFieldComponent.selectedCountryChange
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe((country: SkyCountryData) => {
+        // Write the value again to cause validation to refire
+        this.writeValue(this.modelValue);
+        this.adapterService.setElementPlaceholder(this.elRef.nativeElement, country.placeholder);
+        this.selectedCountryChange.emit(country);
+      });
+    // This is needed to address a bug in Angular 4, where the value is not changed on the view.
+    // See: https://github.com/angular/angular/issues/13792
+    const control = (<NgControl>this.injector.get(NgControl)).control as FormControl;
+    /* istanbul ignore else */
+    if (control && this.modelValue) {
+      control.setValue(this.modelValue, { emitEvent: false });
+      /* istanbul ignore else */
+      if (this.changeDetector) {
+        this.changeDetector.detectChanges();
+      }
+    }
+  }
+
+  /**
+   * Writes the new value for reactive forms on change events on the input element
+   * @param event The change event that was received
+   */
+  @HostListener('change', ['$event'])
+  public onChange(event: any): void {
+    this.writeValue(event.target.value);
+  }
+
+  /**
+   * Marks reactive form controls as touched on input blur events
+   */
+  @HostListener('blur')
+  public onBlur(): void {
+    this._onTouched();
+  }
+
+  /**
+   * Writes the new value for reactive forms
+   * @param value The new value for the input
+   */
+  public writeValue(value: any): void {
+    this.modelValue = value;
+  }
+
+  public registerOnChange(fn: (value: any) => any): void { this._onChange = fn; }
+
+  public registerOnTouched(fn: () => any): void { this._onTouched = fn; }
+
+  public registerOnValidatorChange(fn: () => void): void { this._validatorChange = fn; }
+
+  /**
+   * Sets the disabled state on the input
+   * @param isDisabled the new value of the input's disabled state
+   */
+  public setDisabledState(isDisabled: boolean): void {
+    this.disabled = isDisabled;
+  }
+
+  /**
+   * Validate's the form control's current value
+   * @param control the form control for the input
+   */
+  public validate(control: AbstractControl): { [key: string]: any } {
+    let value = control.value;
+
+    if (!value) {
+      return undefined;
+    }
+
+    if (!this.skyPhoneFieldComponent.validateNumber(value) && !this.skyPhoneFieldNoValidate) {
+      return {
+        'skyPhoneField': {
+          invalid: control.value
+        }
+      };
+    }
+
+    return undefined;
+  }
+
+  /*istanbul ignore next */
+  private _onChange = (_: any) => { };
```

Comment:
Underscores are reserved for `get/set` fields.

It's implied, here, but probably should be written out more specifically: https://docs.blackbaud.com/skyux-team-docs/style#class-properties

Reply:
Ya. I remember us talking about that on your datepicker changes. This predates my learning :) I have changed them but did have to change the name of the host listener functions because of that. Lets chat if you don't like how those look.